code-studio/src/js/components/progress/stage_progress.jsx

```diff
@@ -46,4 +44,6 @@ const StageProgress = React.createClass({
     );
   }
 });
-module.exports = StageProgress;
+export default connect((state, ownProps) => ({
+  levels: ownProps.levels || state.stages[0].levels
```

Comment:
What allows us to be hard-coded to `stages[0]` here?  I'm not familiar with your underlying state, but at a glance I'd expect the index to be variable.